src/coreclr/jit/morph.cpp

```diff
@@ -10869,7 +10869,71 @@ GenTree* Compiler::fgOptimizeHWIntrinsic(GenTreeHWIntrinsic* node)
             INDEBUG(node->gtDebugFlags |= GTF_DEBUG_NODE_MORPHED);
             return node;
         }
+#if defined(TARGET_XARCH)
+        case NI_SSE_And:
+        case NI_SSE2_And:
+        case NI_AVX_And:
+        case NI_AVX2_And:
+        {
+            if (node->GetOperandCount() != 2)
```

Comment:
when exactly it might be not 2 ?

Reply:
Shouldn't ever not be 2. If it was, we'd have a buggy node.

Reply:
Use an assert then instead?